daemon/api.go

```diff
@@ -949,19 +949,28 @@ func changeInterfaces(c *Command, r *http.Request) Response {
 	}
 	switch a.Action {
 	case "connect":
-		err := c.d.interfaces.Connect(a.Plug.Snap, a.Plug.Name, a.Slot.Snap, a.Slot.Name)
+		if len(a.Plugs) != 1 || len(a.Slots) != 1 {
```

Comment:
I think these checks are all wrong :-)

That is: you correctly check for `!= 1`, but you error return assumes `>1`.

Maybe you can check for `>1` outside of the case, return the "not implemented" 400 there, and then check for the `<1` in the cases where you need to check, and return the appropriate "missing" 400 in each one?